cloud: add RBAC config for RBAC-enabled Secure Kubernetes Cluster

acoshift opened the pull request:
Hello, I try to run CockroachDB on RBAC-enabled Kubernetes Cluster. Hope this config can help other who have problem like me too.
This might fixed #20065 too ?

acoshift committed:
cloud: add rbac support to secure kubernetes configuration

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/20116)
<!-- Reviewable:end -->

a-robinson commented:
Thanks for the contribution, @acoshift! I'll take this for a spin soon. And @a6802739, if you have comments or thoughts on this based on what you've done for #20065, I'd appreciate them.

a6802739 commented:
@a-robinson, when I want to tackle with #20065, I just want to learn more about Kubernetes. 
this is great, Thanks!

a-robinson commented on cloud/kubernetes/rbac.yaml:
```diff
@@ -0,0 +1,57 @@
+  name: cockroachdb
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
```
GKE unfortunately requires an extra step before creating this `Role` and the `ClusterRole` is permitted (https://cloud.google.com/kubernetes-engine/docs/how-to/role-based-access-control#prerequisites_for_using_role-based_access_control). I can update the README/docs for this if you don't want to.

a-robinson reviewed (commented):
Thanks a ton for the contribution, @acoshift! I've gone through, learned about, and tried some of this for myself, and it looks like exactly what we need.

I'd prefer to just always use RBAC in secure mode, though. Is there a reason not to? If not, would you mind modifying your commit to just add the `serviceAccountName` to all the existing `-secure.yaml` files and add the contents of `rbac.yaml` to `cockroachdb-statefulset-secure.yaml`? If you're too busy just let me know and I can take care of it.

acoshift commented:
@a-robinson Hello, I've update the yaml. Can you add comments, and prerequisites for GKE ?
My English is very bad 😅

a-robinson committed:
cloud: Rename Container Engine to Kubernetes Engine

Release note: None

a-robinson committed:
cloud: Provide extra command needed to use RBAC on GKE

Release note (general change): The secure Kubernetes instructions now
support running in clusters with strict RBAC enabled.

a-robinson reviewed (approved):
Thanks @acoshift! I've squashed your commits together and added documentation about how to use this on GKE.

I'm going to update our docs site at the same time as this so I won't merge this quite yet. Also cc @mberhault so that I'm not effectively approving my own docs change.

a-robinson committed:
cloud: Add labels to Kubernetes RBAC resources for easier deletion

Release note: None

mberhault commented on cloud/kubernetes/README.md:
````diff
@@ -116,13 +116,37 @@ kubectl create -f cluster-init.yaml
 minikube start --extra-config=controller-manager.ClusterSigningCertFile="/var/lib/localkube/ca.crt" --extra-config=controller-manager.ClusterSigningKeyFile="/var/lib/localkube/ca.key"
 ```
 
+**REQUIRED**: because secure mode creates certificate signing requests and
````
This paragraph is a bit rambly.

a-robinson replied:
Copied the text from my docs change, which was a bit less rambly. Look better?

mberhault commented on cloud/kubernetes/README.md:
````diff
@@ -116,13 +116,37 @@ kubectl create -f cluster-init.yaml
+roles](https://cloud.google.com/kubernetes-engine/docs/how-to/role-based-access-control#prerequisites_for_using_role-based_access_control)
+by running:
+```shell
+kubectl create clusterrolebinding cluster-admin-binding --clusterrole=cluster-admin --user=<your.google.cloud.email@example.org>
````
This is a GKE-specific thing, right? It probably belongs in the `On GCE` section.

a-robinson replied:
Yes, moved.

a-robinson reviewed (commented):
TFTR!

mberhault commented on cloud/kubernetes/README.md:
```diff
@@ -86,12 +86,28 @@ Set up your cluster following the
 
+If you're using GKE and want to run a secure cluster, one extra step is
+required. A limitation in GKE's Role-Based Access Control (RBAC) integration
+necessatitates running a special command in order to let you create the RBAC
```
s/necessaahhh-I-can't-stop-typing/necessitates/

a-robinson replied:
:)

mberhault reviewed (commented):
LGTM

a-robinson commented:
Docs updates done in https://github.com/cockroachdb/docs/pull/2281
